vendor/engines/secure_rooms/app/services/secure_rooms/access_handlers/order_handler.rb

```diff
@@ -34,15 +32,13 @@ def find_or_create_order
         end
       end
 
-      def fulfill_order
-        order_detail.complete!
-      end
-
       def create_order
         ActiveRecord::Base.transaction do
           create_order_and_detail
-          order.validate_order!
-          order.purchase!
+          if occupancy.account_id?
+            order.validate_order!
+            order.purchase!
```

Comment:
I suspect this will get weird when we get to the scanning out without an in because you can't purchase an order without an id, but you also probably won't see unpurchased orders in the problems tab. We can deal with that later, though.